app/middlewares/album.js

```diff
@@ -34,3 +34,24 @@ exports.uniqueAlbumBoughtValidation = (req, res, next) => {
     }
   });
 };
+
+exports.retrieveUserAlbums = (req, res, next) => {
+  const userToFind = { userId: req.params.user_id };
+  logger.info(`Attempting to find all Albums of user with id: ${userToFind.userId}.`);
+  Album.getAlbumList(userToFind)
```

Comment:
I don't think this should be done on the middleware, you should do it directly on the controller. Also, keep in mind that you are always retrieving the album list before checking if the user is actually allowed to ask for it. If you check the user's permission's first, you can avoid making that request to the database.